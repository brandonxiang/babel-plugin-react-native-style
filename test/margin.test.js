const pluginTester = require('babel-plugin-tester');
const reactNativeStyle = require('../src/index');

pluginTester({
  plugin: reactNativeStyle,
  pluginName: 'babel-plugin-react-native-style',

  tests: [
    {
      snapshot: true,
      title: 'margin test',
      pluginOptions: { rpx: true, designWidth: 750 },
      code: `
      const styles = stylesheet.create({
        one: { margin: 0 },
        two: { margin: '10 1' },
        three: { margin: '10 1 5' },
        four: { margin: '10 1 5 3' },
        five: { margin: '10 auto 5' },
      })
    `,
    },
  ],
});
