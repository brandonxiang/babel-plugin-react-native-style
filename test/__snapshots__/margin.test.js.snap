// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-react-native-style margin test: margin test 1`] = `
"
const styles = stylesheet.create({
  one: { margin: 0 },
  two: { margin: '10 1' },
  three: { margin: '10 1 5' },
  four: { margin: '10 1 5 3' },
  five: { margin: '10 auto 5' },
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { __RPX } from 'babel-plugin-react-native-style/src/runtime/index';
const styles = stylesheet.create({
  one: { margin: __RPX(0, 750) },
  two: {
    marginTop: __RPX(10, 750),
    marginRight: __RPX(1, 750),
    marginBottom: __RPX(10, 750),
    marginLeft: __RPX(1, 750)
  },
  three: {
    marginTop: __RPX(10, 750),
    marginRight: __RPX(1, 750),
    marginBottom: __RPX(5, 750),
    marginLeft: __RPX(1, 750)
  },
  four: {
    marginTop: __RPX(10, 750),
    marginRight: __RPX(1, 750),
    marginBottom: __RPX(5, 750),
    marginLeft: __RPX(3, 750)
  },
  five: {
    marginTop: __RPX(10, 750),
    marginRight: 'auto',
    marginBottom: __RPX(5, 750),
    marginLeft: 'auto'
  }
});
"
`;
